gui/src/app/RunPanel/RunPanel.tsx

```diff
@@ -24,6 +24,8 @@ const RunPanel: FunctionComponent<RunPanelProps> = ({ width, height, compiledUrl
     const [runStatus, setRunStatus] = useState<RunStatus>('waiting');
     const [errorMessage, setErrorMessage] = useState<string>('');
     const [samples, setSamples] = useState<number[][] | undefined>(undefined);
+    const [paramNames, setParamNames] = useState<string[] | undefined>(undefined);
```

Comment:
Apart from conditioning run doneness on `paramNames` being non-undefined (l. 147), any reason not to keep this as a simple `string[]` and use an empty list for the default?

Reply:
No, and we could probably do the same for `samples`.